app/models/conversion_host.rb

```diff
@@ -4,4 +4,105 @@ class ConversionHost < ApplicationRecord
   acts_as_miq_taggable
 
   belongs_to :resource, :polymorphic => true
+
+  def active_tasks
+    ServiceTemplateTransformationPlanTask.where(:state => 'active').select do |task|
+      task.conversion_host == self
```

Comment:
We should move this to a relation so that we don't need to loop over tasks and it looks like `.conversion_host` is itself a different db query with an options hash.

Reply:
How do we do that ? Simply add `has_many :service_template_transformation_plan_task` ?

Reply:
The relation between conversion_host and tasks is probably not that trivial. Currently, the ServiceTemplateTransformationPlanTask class is a sub-sub-class of MiqRequestTask, so the relationship would imply add a `conversion_host_id` in `miq_request_tasks`.

From a semantic point of view, this wouldn't make sense, as the conversion host is specific to ServiceTemplateTransformationPlanTask. But technically, it's pretty easy and can be safely ignore by other sub-classes of MiqRequestTask.

What do you think @agrare ?

Reply:
@fdupont-redhat for starters this [`ServiceTemplateTransformationPlanTask`](https://github.com/ManageIQ/manageiq/blob/master/app/models/service_template_transformation_plan_task.rb#L49-L51) should be `belongs_to :conversion_host` and yes we'd need to add a schema reference to the base class to do that.

Then this just becomes `ConversionHost has_many :service_template_transformation_plan_tasks` (_just_ haha that's a mouthful)

> From a semantic point of view, this wouldn't make sense, as the conversion host is specific to ServiceTemplateTransformationPlanTask

yeah that's how STI works, all columns for all sub-classes go in the same table.  I wish there was a better way

This way is going to perform really badly compared to a pure db query approach, though I guess no worse than the existing automate approach.

If we need to keep this for backport purposes I'm okay with that but lets put in the schema changes to get this right going forward.

Reply:
@agrare Here is the PR for schema change: https://github.com/ManageIQ/manageiq-schema/pull/281.
Anyway, we're too late for Hammer, aren't we ?

Reply:
Yes we had a window after they branched but before the first build but afaik @simaishi has cut a beta-1 tag

Reply:
Okay @fdupont-redhat now that the schema change is in we can do this with a has_many and if you add a scope on `:state => "active"` to `MiqRequestTask` we can do the active tasks with `-> { active }`